sstables/sstables_manager.cc

```diff
@@ -169,6 +169,41 @@ void sstables_manager::increment_total_reclaimable_memory_and_maybe_reclaim(ssta
     smlogger.info("Reclaimed {} bytes of memory from SSTable components. Total memory reclaimed so far is {} bytes", memory_reclaimed, _total_memory_reclaimed);
 }
 
+future<> sstables_manager::maybe_reload_reclaimed_components() {
+    size_t memory_reclaim_threshold = _available_memory * _db_config.components_memory_reclaim_threshold();
+    size_t memory_available_for_reclaimable_components = memory_reclaim_threshold - _total_reclaimable_memory;
+
```

Comment:
This variable becomes out-of-date as soon as you co_await. Better to always compute from live data and not cache anything.